src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java

```diff
@@ -126,60 +120,131 @@ public void publishDiagnostics(PublishDiagnosticsParams publishDiagnosticsParams
     public void showMessage(MessageParams messageParams) {
         String title = "Language Server message";
         String message = messageParams.getMessage();
-        ApplicationUtils.invokeLater(() -> {
-            MessageType msgType = messageParams.getType();
+
+        if (isModal) {
+            ApplicationUtils.invokeLater(() -> {
+                MessageType msgType = messageParams.getType();
+                switch (msgType) {
+                    case Warning:
+                        Messages.showWarningDialog(message, title);
+                        break;
+                    case Error:
+                        Messages.showErrorDialog(message, title);
+                        break;
```

Comment:
Shall we consider swapping `Warning` and `Error` cases, just be aligned with the java log level standards?